src/main/io/vtx_control.c

```diff
@@ -0,0 +1,247 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+// Get target build configuration
+#include "platform.h"
+
+#include "common/maths.h"
+
+#include "config/config_eeprom.h"
+#include "config/parameter_group.h"
+#include "config/parameter_group_ids.h"
+
+#include "drivers/vtx_common.h"
+#include "drivers/buttons.h"
+#include "drivers/light_led.h"
+#include "drivers/system.h"
+
+#include "fc/config.h"
+#include "fc/runtime_config.h"
+
+#include "io/beeper.h"
+#include "io/osd.h"
+#include "io/vtx_control.h"
+
+
+#if defined(VTX_CONTROL) && defined(VTX_COMMON)
+
+PG_REGISTER(vtxConfig_t, vtxConfig, PG_VTX_CONFIG, 1);
+
+static uint8_t locked = 0;
+
+void vtxControlInit(void)
+{
+    // NOTHING TO DO
+}
+
+static void vtxUpdateBandAndChannel(uint8_t bandStep, uint8_t channelStep)
+{
+    if (ARMING_FLAG(ARMED)) {
+        locked = 1;
+    }
+
+    if (!locked) {
+        uint8_t band = 0, channel = 0;
+        vtxCommonGetBandChan(&band, &channel);
+        vtxCommonSetBandChan(band + bandStep, channel + channelStep);
+    }
+}
+
+void vtxIncrementBand(void)
+{
+    vtxUpdateBandAndChannel(+1, 0);
+}
+
+void vtxDecrementBand(void)
+{
+    vtxUpdateBandAndChannel(-1, 0);
+}
+
+void vtxIncrementChannel(void)
+{
+    vtxUpdateBandAndChannel(0, +1);
+}
+
+void vtxDecrementChannel(void)
+{
+    vtxUpdateBandAndChannel(0, -1);
+}
+
```

Comment:
that's not really possible right now, both need to be set at the same time due to the implementations of VTX hardware. In the case of the RTC6705 band and channel are ALWAYS set at the same time because a frequency is what the hardware needs.  It would be BAD to change band and then change channel because you might briefly transmit on a frequency someone else needs.

You could add the complexity of queuedBand and queuedChannel and then if the complement channel or band is changed then calculate new frequency once both setChannel and setBand are called but then you also need to handle only setChannel or setBand being called and you'll end up in a right mess 😄

Keeping as-is for now.